refresh.template.py

```diff
@@ -147,6 +148,20 @@ def _get_headers_msvc(compile_args: typing.List[str], source_path: str):
         # End:   template filled by Bazel
     ))
 
+    # Write header_cmd to a temporary file, so we can use it as a parameter file to cl.exe,
+    # because Windows cmd has a limitation of 8191 charactors. So here we make a threshold of len(compile_args) < 8000
+    # See https://docs.microsoft.com/en-us/troubleshoot/windows-client/shell-experience/command-line-string-limitation
+    WIN_CMD_LIMITS = 8000
+    tempdir = None
+    if len('\n'.join(header_cmd)) >= WIN_CMD_LIMITS:
+        tempdir = tempfile.TemporaryDirectory()
```

Comment:
Can I ask why we're doing a TemporaryDirectory instead of, e.g., a context manager over  tempfile.NamedTemporaryFile? Like for example:
`with tempfile.NamedTemporaryFile(...`

(Offhand, I think that'd be simpler than creating and manually destroying a directory, too, but maybe I'm missing something.)

Reply:
Oh, got your point.
Just test and `subprocess.call` will return error code **1**, with an stderr output "_command line too long_".

We may check return code 1 or stderr text (the text is in locale language)

Reply:
Will change to NamedTemporaryFile